eureka2-write-server/src/main/java/com/netflix/eureka2/server/service/bootstrap/PeerRegistryBootstrapService.java

```diff
@@ -0,0 +1,118 @@
+package com.netflix.eureka2.server.service.bootstrap;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.LinkedHashSet;
+
+import com.netflix.eureka2.Server;
+import com.netflix.eureka2.interests.ChangeNotification;
+import com.netflix.eureka2.interests.ChangeNotifications;
+import com.netflix.eureka2.interests.Interests;
+import com.netflix.eureka2.registry.Source;
+import com.netflix.eureka2.registry.SourcedEurekaRegistry;
+import com.netflix.eureka2.registry.instance.InstanceInfo;
+import com.netflix.eureka2.server.resolver.EurekaEndpoint;
+import com.netflix.eureka2.server.resolver.EurekaEndpointResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Notification;
+import rx.Observable;
+import rx.Scheduler;
+import rx.functions.Func1;
+import rx.schedulers.Schedulers;
+
+/**
+ * @author Tomasz Bak
+ */
+@Singleton
+public class PeerRegistryBootstrapService implements RegistryBootstrapService {
```

Comment:
Nitpick, peer in the name here assumes bootstrapping from peer write nodes, whereas we can bootstrap from any arbitrary eureka source based on configuration.